src/components/TimeSeriesCard/TimeSeriesCard.jsx

```diff
@@ -93,21 +94,37 @@ export const determinePrecision = (size, value, defaultPrecision) => {
  * @param {array} series, an array of lines to create in our chart
  * @param {array} values, the array of values from our data layer
  *
+ * TODO: Handle empty data lines gracefully and notify the user of data lines that did not
+ * match the dataFilter
+ *
  * @returns {object} with a labels array and a datasets array
  */
 export const formatChartData = (timeDataSourceId = 'timestamp', series, values) => {
-  return {
-    labels: [...new Set(values.map(val => val[timeDataSourceId]))],
-    datasets: series.map(({ dataSourceId, dataFilter = {}, label, color }) => ({
-      label,
-      ...(color ? { fillColors: [color] } : {}),
-      data:
-        filter(values, dataFilter) &&
-        filter(values, dataFilter) // have to filter out null values from the dataset, as it causes Carbon Charts to break
-          .filter(i => !isNil(i[dataSourceId]))
-          .map(i => ({ date: new Date(i[timeDataSourceId]), value: i[dataSourceId] })),
-    })),
-  };
+  const timestamps = [...new Set(values.map(val => val[timeDataSourceId]))];
```

Comment:
we really need a testcase to cover this function

Reply:
Added